### PR TITLE
Bump netty for CVE-2024-47535

### PR DESCRIPTION
**Pull request by russwyte**
Addresses CVE alert I have been getting from github dependabot.

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-47535

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ import sbt.*
 
 object Dependencies {
   val JwtCoreVersion                = "10.0.1"
-  val NettyVersion                  = "4.1.112.Final"
+  val NettyVersion                  = "4.1.115.Final"
```
Comment (jivanic-demystdata) [inline project/Dependencies.scala:5]:
```suggestion
  val NettyVersion                  = "4.1.116.Final"
```

Comment (guizmaii) [inline project/Dependencies.scala]:
```suggestion
  val NettyVersion                  = "4.1.116.Final"
```

**Review by guizmaii — APPROVED** [review]

**Comment by guizmaii** [comment]
@russwyte I updated to `4.1.116` and it seems that we're using a deprecated method, making the CI fail. Could you have a look, please?

**Comment by russwyte** [comment]
> @russwyte I updated to `4.1.116` and it seems that we're using a deprecated method, making the CI fail. Could you have a look, please?

Sure - I will look.

**Comment by russwyte** [comment]
@guizmaii Fortunately it was easy fix - just a deprecated method. 👍

File: zio-http/jvm/src/main/scala/zio/http/netty/server/ServerSSLDecoder.scala [hunk 2]
```diff
@@ -131,7 +131,7 @@ private[zio] class ServerSSLDecoder(sslConfig: SSLConfig, cfg: Server.Config) ex
     val httpBehaviour = sslConfig.behaviour
     if (in.readableBytes < 5)
       ()
-    else if (SslHandler.isEncrypted(in)) {
+    else if (SslHandler.isEncrypted(in, false)) {
```
Comment (guizmaii) [inline zio-http/jvm/src/main/scala/zio/http/netty/server/ServerSSLDecoder.scala:134]: @russwyte Just for my information, what is this `false` for?

Found the PR making the change: https://github.com/netty/netty/pull/14243

image

**Review by guizmaii — APPROVED** [review]
LGTM

**Comment by russwyte** [comment]
I just went with the strongly suggested option.

**Review by 987Nabil — APPROVED** [review]

**Pull request merged**